### DISCUSSION TITLE
Arraxis Merge - A powerful tool for comparison tasks (files and directories)

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#4082] [Show & Tell]
Recently I had to migrate to OpenMage a big Magento v1.9.2.4 shopping cart intensively customized for 8 years. The changes were made more or less carefully by various people. I found a lot even in the Mage core files.

Fortunately, I used a program that helped me enormously to complete an analysis work that required a lot of attention. This is called Arraxis Merge, from there https://www.araxis.com/merge/index.en.

I initially started from analyzing v1.9.2.4 with v1.9.2.4 customized to understand all changes. I deleted everything which was in common. After cleaning, I copied the latest OM v20 then I updated all locale files, templates and so one. The fact that Arraxis allows viewing line by line helped me enormously, because I was able to copy or remove the differences from one side to another. I revised the process several times and then took the updated version to a DDEV test environment with the store database. There were some small issues related to deprecated errors in extensions, but it runs incredibly well with PHP 8.3. I will spend more time testing the Frontend/Backend functionality before moving it to production, but so far everything has gone unusually well. There are no errors in the browser console or in the log files.

I would like to thank you all who contributed to OpenMage. If this website lasted 8 years based on a platform from the past, it will last even longer with the transition to the latest version of OpenMage and PHP 8.3.

I would have one comment, the layout/template system that Magento team created 16 years ago is not at all easy to use, but those who get familiar with it can use it. I would have preferred to have something like in OpenCart where customizing the frontend is easier to do, using Twig that is easy to look over the code, but that is another story.

**Comment by fballiano** [comment]
nice story, tough job considering the super old M1 version you were starting with. Thanks for sharing :-)

The M1 platform truly is still awesome ;-)

**Comment by empiricompany** [comment]
Great work @ADDISON74! 
I love hearing about these success stories. I imagine that moving to another platform would have involved much higher costs!

Personally i use this two command from https://github.com/AOEpeople/mpmd a plugin for n98-magerun:
- mpmd:codepooloverrides: Find all code pool overrides
- mpmd:corehacks: Find all core hacks

to check and sanitize all installations coming from new customers

PS: a good alternative to Arraxis Merge is https://github.com/winmerge/winmerge for windows

**Reply by addison74** [reply]
Thank you for the appreciation. 

I analyzed the migration to Magento 2, WooCommerce and OpenCart, but it involves a lot of work. On the other hand the transition from Magento 1 to OpenMage was the best option, in addition to the changes made here and maintaining the code is still a solid solution. 

I preferred to analyze everything, even if it cost me time to see what I would add, delete. change at such a big store, letting a script do tasks automatically it could be not a safe solution. I know about WinMerge, but Arraxis, even if it is expensive, has many more features and it is extremely fast when analyzing tens of thousands of files.

OpenMage is a very good platform for anyone, but they wanted to forget it because there were financial interests in the middle. In my opinion, Magento 2 has not reached the level of popularity of Magento 1.

**Reply by SuSiSonnenschein111** [reply]
+1 for mpmd
